lib/galaxy/config/schemas/config_schema.yml

```
@@ -4104,3 +4104,21 @@ mapping:
         per_host: true
         desc: |
           Enable the integration of the Galaxy Help Forum in the tool panel. This requires the help_forum_api_url to be set.
+
+      file_source_temp_dir:
+        type: str
+        required: false
+        desc: |
+          Directory to store temporary files for file sources. This defaults to new_file_path if not set.
+      
+      file_source_webdav_use_temp_files:
+        type: bool
+        default: false
+        desc: |
+          Default value for use_temp_files for webdav plugins that don't explicitly declare this.
+
+      file_source_s3fs_listings_expiry_time:
+        type: int
+        default: 60
+        desc: |
+          Default value for listings_expiry_time for s3fs file source plugins that don't explicitly declare this.
```

```suggestion
      file_source_listings_expiry_time:
        type: int
        default: 60
        desc: |
          Number of seconds before file source content listings are refreshed. Shorter times will result in more queries while browsing a file sources. Longer times will result in fewer requests to file sources but outdated contents might be displayed to the user.
```
This is more reasonable to make configurable, but it still seems like a setting no-one should need to touch. I can also see that being something we might have to implement for other file sources if request volume becomes problematic, hence dropping the s3fs from the setting name.

---

It only affects s3fs - I feel like changing it makes it more likely people will try to fix other timeouts by adjusting it. I will make this change but I'm grumpy about it.

---

You could add `currently only affects s3fs file source` ?

---

(Updated the suggestion with that)